lib/generators/has_friendship/has_friendship_generator.rb

```diff
@@ -9,7 +9,10 @@ def self.source_root
   end
 
   def self.next_migration_number(path)
-    Time.now.utc.strftime("%Y%m%d%H%M%S")
+    next_num = Time.now.utc.strftime("%Y%m%d%H%M%S").to_i
+    current_num = current_migration_number(path)
+    next_num += (current_num - next_num + 1) if current_num >= next_num
```

Comment:
How you propose it to be, is one of the ways I had it at first, but then I realized that current_num CAN be greater than next_num.  Remember, next_num is basically the timestamp when you call the method, and that method can be called many times per second.  So over 10 calls, they can all have the same next_num.  But current_num will increment with every call.  So current_num can be greater than next_num.  And because of that, you have to do it how I propose - figuring the difference between next and current (or something like it.)

And as far as what Pensarfeo said, the sleep(1) solution was the first one I thought of, but it just seems wrong to make your gem install slower, and also, the only way there would be a problem is if another gem install or migration generator was run a second or two after this one finishes, which doesn't seem likely. And really, then you just go and manually change the numbers. Which really could be said for this gem install - leave it alone and let people manually change the numbers if it doesn't work. But I almost didn't use this gem because of the error.  I'm glad I ended up looking into it, because I'm using it and it's fantastic.

So I still say that this pull request is the way to go.  Thanks!